features/0509-action-menu/README.md

```diff
@@ -0,0 +1,211 @@
+
+The protocol can be initiated by either the `requester` asking for the root menu or the `responder` sending an unsolicited root menu.  The protocol ends when the `requester` issues a `perform` operation or an internal timeout on the `responder` causes it to discard menu context.  At any time a `requester` can reset the protocol by requesting the root menu from a `responder`.
+
+Whilst the protocol is defined here as uni-directional (i.e `requester` to `responder`), both Agents may support both `requester` and `responder` roles simultaneously. Such cases would result in two instances of the action-menu protocol operating in parrallel. 
+
+### Roles
+
+There are two roles in the action-menu protocol: `requester` and `responder`.
+
+The `requester` asks the `responder` for menu definitions, presents them to a user, and intitates subsequent action items from the menu through further requests to the `responder`.
+
+The `responder` presents an initial menu definition containing actionable elements to a `requestor` and then responds to subsequent action requests from the menu.  
+
+### States
+
+[![state machines](state-machines.png)](https://docs.google.com/spreadsheets/d/1KZ78KDxmXdck068aJpn-7BbDNfeaTcUyDgk6b2efCxc/edit?usp=sharing)
+
+#### States for Requester
+
+State&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;| Description
+------ | -----------
+null | No _menu_ has been requested or received
+awaiting-root-menu | _menu-request_ message has been sent and awaiting root _menu_ response
+preparing-selection | _menu_ message has been received and a user selection is pending
+done | _perform_ message has been sent and protocol has finished. Perform actions can include requesting a new menu which will re-enter the state machine with the receive-menu event from the null state.
+
+#### States for Responder
+
+State&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;| Description
+------ | -----------
+null | No _menu_ has been requested or sent
+preparing-root-menu | _menu-request_ message has been received and preparing _menu_ response for root menu
+awaiting-selection | _menu_ message has been sent and are awaiting a _perform_ request
+done | _perform_ message has been received and protocol has finished. Perform actions can include requesting a new menu which will re-enter the state machine with the send-menu event from the null state.
+
+### Messages
+
+#### menu
+
+A `requestor` is expected to display only one active _menu_ per connection when action menus are employed by the `responder`. A newly received _menu_ is not expected to interrupt a user, but rather be made available for the user to inspect possible actions related to the `responder`.
+
+```jsonc
+{
+  "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/action-menu/1.0/menu",
```

Comment:
There is a new form for types --- https://didcomm.org/action-menu/%VER/menu

The Version of the protocol (1.0, in this case) should be defined just after the header and then referenced with the "%VER".  That's so we only have one copy of the version.

Reply:
Done